Tests/CommandsTests/PackageCommandTests.swift

```diff
@@ -1152,6 +1151,58 @@ class PackageCommandTestCase: CommandsBuildProviderTestCase {
         }
     }
 
+    func testPackageAddTargetWithoutModuleSourcesFolder() async throws {
+        try await testWithTemporaryDirectory { tmpPath in
+            let fs = localFileSystem
+            let manifest = tmpPath.appending("Package.swift")
+            try fs.writeFileContents(manifest, string:
+                """
+                // swift-tools-version: 5.9
+                import PackageDescription
+                let package = Package(
+                    name: "SimpleExecutable",
+                    targets: [
+                        .executableTarget(name: "SimpleExecutable"),
+                    ]
+                )
+                """
+            )
+
+            let sourcesFolder = tmpPath.appending("Sources")
+            try fs.createDirectory(sourcesFolder)
+
+            try fs.writeFileContents(sourcesFolder.appending("main.swift"), string:
+                """
+                print("Hello World")
+                """
+            )
+
+            _ = try await execute(["add-target", "client"], packagePath: tmpPath)
+
+            XCTAssertFileExists(manifest)
+            let contents: String = try fs.readFileContents(manifest)
+
+            XCTAssertMatch(contents, .contains(#"targets:"#))
+            XCTAssertMatch(contents, .contains(#".executableTarget"#))
+            XCTAssertMatch(contents, .contains(#"name: "client""#))
+
+            let fileStructure = try fs.getDirectoryContents(sourcesFolder)
+            XCTAssertEqual(fileStructure, ["SimpleExecutable", "client"])
+            XCTAssertTrue(fs.isDirectory(sourcesFolder.appending("SimpleExecutable")))
+            XCTAssertTrue(fs.isDirectory(sourcesFolder.appending("client")))
+            XCTAssertEqual(try fs.getDirectoryContents(sourcesFolder.appending("SimpleExecutable")), ["main.swift"])
+            XCTAssertEqual(try fs.getDirectoryContents(sourcesFolder.appending("client")), ["client.swift"])
+        }
+    }
+
+    func testAddTargetWithoutManifestThrows() async throws {
```

Comment:
suggestion (possibly-blocking): Although the current tests are great in validating the file system structure, can we add a test that validate the Package can still be built, and maybe even ensure the tests pass, if we run `swift package add-target foo` on an package containing a single source?